### PR TITLE
Feature/type orm configuration

### PR DESCRIPTION
**Pull request by CristoferPortela**
typeorm basic configurations added

### CODE REVIEW

File: apps/api/src/app/app.module.ts [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-import { AppResolver } from './app.resolver';
-import { TerminusModule } from '@nestjs/terminus';
+import {AuthModule} from '../auth/auth.module';
+import {UsersModule} from '../users/users.module';
+import {WebsitesModule} from '../websites/websites.module';
+import {GraphQLModule} from '@nestjs/graphql';
+import {ApolloDriver, ApolloDriverConfig} from '@nestjs/apollo';
+import {join} from 'path';
+import {AppResolver} from './app.resolver';
+import {TerminusModule} from '@nestjs/terminus';
```
Comment (vygandas) [inline apps/api/src/app/app.module.ts:15]: How come there're no spaces after { and before } but prettier still passed 🧐 It should be formatted as everything else in the code. Can it be a configuration issue?

File: apps/api/src/app/app.module.ts [hunk 2]
```diff
@@ -23,7 +23,8 @@ import { TerminusModule } from '@nestjs/terminus';
     TypeOrmModule.forRootAsync({
       imports: [ConfigModule],
       useFactory: (configService: ConfigService) =>
-        configService.get('database'),
+        configService.get('database')
+      ,
```
Comment (vygandas) [inline apps/api/src/app/app.module.ts:27]: Why is this one here?

File: apps/api/src/config/database.ts [hunk 3]
```diff
@@ -1,39 +1,29 @@
...
-  type: 'postgres',
-  host: process.env.POSTGRES_HOST,
-  port: parseInt(process.env.POSTGRES_PORT, 10) || 5432,
-  username: process.env.POSTGRES_USER,
-  password: process.env.POSTGRES_PASSWORD,
-  database: process.env.POSTGRES_DB,
-  ssl: false,
-  entities: [join(__dirname, '../**/*.entity{.ts,.js}')],
-  // We are using migrations, synchronize should be set to false.
+  type: process.env.TYPEORM_DRIVER as never,
```
Comment (vygandas) [inline apps/api/src/config/database.ts:6]: `never` shouldn't be used here. Especially when there's a specific interface `PostgresConnectionOptions` used which has only one type variant. Should be either an appropriate interface used or multiple interfaces with `|`

File: apps/api/src/config/datasource.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import dotenv from 'dotenv';
```
Comment (vygandas) [inline apps/api/src/config/datasource.ts:1]: Maybe Datasource can import configuration from the database config? We should avoid duplicated code.

File: package.json [hunk 5]
```diff
@@ -55,6 +58,7 @@
     "graphql": "^16.6.0",
     "helmet": "^6.0.0",
     "mailgun.js": "^8.0.2",
+    "mysql2": "^2.3.3",
```
Comment (vygandas) [inline package.json:61]: Is it the user anywhere in the code? If it's not imported anywhere, let's remove this.

**Review by vygandas — CHANGES_REQUESTED** [review]

File: apps/api/src/app/app.module.ts (hunk 1 above)
Comment (CristoferPortela) [reply apps/api/src/app/app.module.ts:15]: probably my IDE had not configured the correct prettier here, I'll change it and commit

Comment (vygandas) [reply apps/api/src/app/app.module.ts:15]: Also please make sure `yarn prettier` fails if it's left like that for the future :)

Comment (CristoferPortela) [reply apps/api/src/app/app.module.ts:15]: The prettierc has no rule about this, I'll change that then

File: apps/api/src/app/app.module.ts (hunk 2 above)
Comment (CristoferPortela) [reply apps/api/src/app/app.module.ts]: Solved

File: apps/api/src/config/database.ts (hunk 3 above)
Comment (CristoferPortela) [reply apps/api/src/config/database.ts]: fixed

File: package.json (hunk 5 above)
Comment (CristoferPortela) [reply package.json]: removed

**Review by vygandas — APPROVED** [review]
Now looks good 🙂

**Pull request merged**